instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/__init__.py

```diff
@@ -131,16 +134,20 @@ def _instrument(self, **kwargs):
                 ``engine``: a SQLAlchemy engine instance
                 ``engines``: a list of SQLAlchemy engine instances
                 ``tracer_provider``: a TracerProvider, defaults to global
+                ``engine_tracer_options``: a EngineTracerOptions to pass to the engine tracer
```

Comment:
This change is not backward compatible and is going to break user of this instrumentation. It's also breaking convention from all other instrumentation library.

An actual improvement would be to document the actual existing option and keep them at the root level of kwargs. You can then build a dictionnary of options in the function and pass this dictionnary around.

Reply:
So you mean instead of passing a `EngineTracerOptions` in the kwargs we would pass the arguments to build the instance like :
`enable_commenter`
`commenter_options`
`before_cursor_execute_callback` ?

Reply:
yes